zilliqa-macros/src/test.rs

```diff
@@ -55,22 +55,33 @@ pub(crate) fn test_macro(_args: TokenStream, item: TokenStream) -> TokenStream {
                     let subscriber = tracing_subscriber::fmt()
                         .with_ansi(false)
                         .with_env_filter(tracing_subscriber::EnvFilter::from_default_env());
-                    let _guard = tracing_subscriber::util::SubscriberInitExt::set_default(subscriber);
-
-                    let mut rng = <rand_chacha::ChaCha8Rng as rand_core::SeedableRng>::seed_from_u64(seed);
-                    let network = crate::Network::new(std::sync::Arc::new(std::sync::Mutex::new(rng)), 4, seed);
 
-                    // Call the original test function, wrapped in `catch_unwind` so we can detect the panic.
-                    let result = futures::FutureExt::catch_unwind(std::panic::AssertUnwindSafe(
-                        zilliqa::time::with_fake_time(#inner_name(network))
-                    )).await;
+                    let _guard = tracing_subscriber::util::SubscriberInitExt::set_default(subscriber);
 
-                    match result {
-                        Ok(()) => {},
-                        Err(e) => {
-                            std::panic::resume_unwind(e);
+                    // Optionally print the seed in each tests logging so it can be filtered out later.
+                    use tracing::Instrument;
+                    let span = if std::env::var_os("ZQ_TEST_SEED_SPANS").is_some() {
+                        tracing::span!(tracing::Level::INFO, "", seed)
+                    } else {
+                        tracing::span!(tracing::Level::INFO, "")
+                    };
```

Comment:
Any reason not to just print the seed unconditionally? If I understand correctly, this will only be printed when you have `RUST_LOG=it=info` anyway?

Also should we give the span a name? e.g. `"test"`

Reply:
The seeds are long and ugly so I don't think you would usually want them. Likewise for naming the test span - I did have this before but it doesn't add anything. If you put the network construction in the span too then every single trace message would have 'test' - which adds 0 net information.